src/inspector/models/score/scoreaccessibilitysettingsmodel.cpp

```diff
@@ -0,0 +1,216 @@
+{
+    muse::io::path_t basePath = globalConfiguration()->appDataPath() + "styles/MSN/";
+    switch (preset) {
+    case ScoreStylePreset::DEFAULT:
+        return engravingConfiguration()->defaultStyleFilePath();
+    case ScoreStylePreset::MSN_16MM:
+        return basePath + "16mm_MSN.mss";
+    case ScoreStylePreset::MSN_18MM:
+        return basePath + "18mm_MSN.mss";
+    case ScoreStylePreset::MSN_20MM:
+        return basePath + "20mm_MSN.mss";
+    case ScoreStylePreset::MSN_22MM:
+        return basePath + "22mm_MSN.mss";
+    case ScoreStylePreset::MSN_25MM:
+        return basePath + "25mm_MSN.mss";
+    default:
+        return muse::io::path_t();
+    }
+}
+
+QVariantList ScoreAccessibilitySettingsModel::possibleScoreStylePresets() const
+{
+    QVariantList presets;
+
+    QString text = "text";
+    QString value = "value";
+    QString preset = "preset";
+    QString edited = "edited";
+
+    for (int i = 0; i < static_cast<int>(ScoreStylePreset::MAX_PRESET); ++i) {
+        ScoreStylePreset presetEnum = static_cast<ScoreStylePreset>(i);
+        QString presetName = TConv::translatedUserName(presetEnum);
+
+        presets.append(QVariantMap {
+            { text, presetName },
+            {
+                value,
+                QVariantMap {
+                    { preset, i },
+                    { edited, false }
+                }
+            }
+        });
+
+        if (i == static_cast<int>(m_scoreStylePreset) && m_scoreStylePresetEdited) {
+            QString editedPresetName = muse::qtrc("inspector", "%1 (edited)").arg(presetName);
+            presets.append(QVariantMap {
+                { text, editedPresetName },
+                {
+                    value,
+                    QVariantMap {
+                        { preset, i },
+                        { edited, true }
+                    }
+                }
+            });
+        }
+    }
+
+    return presets;
+}
+
+void ScoreAccessibilitySettingsModel::setScoreStylePresetIndex(int index)
+{
+    int oldIndex = scoreStylePresetIndex();
+
+    if (index == oldIndex) {
+        return;
+    }
+
+    if (m_scoreStylePresetEdited && index > oldIndex) {
+        --index;
+    }
+
+    auto selectedPreset = static_cast<ScoreStylePreset>(index);
+
+    bool presetChanged = (selectedPreset != m_scoreStylePreset);
+    bool editedChanged = (m_scoreStylePresetEdited != false);
+
+    MStyle& style = globalContext()->currentNotation()->elements()->msScore()->score()->style();
+
+    if (presetChanged) {
+        m_scoreStylePreset = selectedPreset;
+        loadStyle(selectedPreset);
+        style.setPreset(selectedPreset);
+
+        emit scoreStylePresetIndexChanged();
+    }
+
+    if (editedChanged) {
+        m_scoreStylePresetEdited = false;
+        style.setPresetEdited(false);
+
+        emit possibleScoreStylePresetsChanged();
+        emit scoreStylePresetIndexChanged();
```

Comment:
You emit `scoreStylePresetIndexChanged()`  twice in this function. That should never happen.

Emitting will cause connected slots to fire (i.e. it will make something happen in the UI), which could potentially be costly in terms of performance. You don't want to emit things more often than you need to.

Normally, it would be better to add a separate conditional block at the end of the function:

```C++
if (presetChanged || editedChanged) {
    emit scoreStylePresetIndexChanged();
}
```

Except in this case you already know that `index != oldIndex`, so you can do this outside of a conditional block:

```C++
emit scoreStylePresetIndexChanged();
```

That should be the final line of this function.